File: src/main/java/at/deckweiss/logging/LogParams.java
package at.deckweiss.logging;

import java.util.ArrayList;
import java.util.List;

public class LogParams {
    private final List<LogParam> logParams;

    public LogParams() {
        this.logParams = new ArrayList<>();
    }

    public LogParams(String name, Object value) {
        this();
        this.logParams.add(new LogParam(name, value));
    }

    public LogParams add(String name, Object value) {
        this.logParams.add(new LogParam(name, value));
        return this;
    }

    public LogParams addIfNotNull(String name, Object value) {
        if (value != null) {
            this.logParams.add(new LogParam(name, value));
        }

        return this;
    }

    LogParam[] toArray() {
        return this.logParams.toArray(new LogParam[] {});
    }
}

File: src/main/kotlin/at/deckweiss/logging/LogParams.kt
package at.deckweiss.logging

class LogParams() {
    private val logParams: MutableList<LogParam> = mutableListOf()

    constructor(name: String, value: Any?) : this() {
        logParams.add(LogParam(name, value))
    }

    fun add(name: String, value: Any?): LogParams {
        logParams.add(LogParam(name, value))
        return this
    }

    fun addIfNotNull(name: String, value: Any?): LogParams {
        value?.let {
            logParams.add(LogParam(name, it))
        }
        return this
    }

    fun toParamString(): String {
        return when {
            logParams.isEmpty() -> ""
            else -> logParams.joinToString(prefix = "[", postfix = "]", separator = ", ") { it.toString() }
        }
    }

}